vxc_integration_test.go

```diff
@@ -122,6 +122,25 @@ func (suite *VXCIntegrationTestSuite) TestVXCBuy() {
 	}
 	serviceKeyID := serviceKeyRes.ServiceKeyUID
 
+	aEndVLAN := GenerateRandomVLAN()
+	bEndVLAN := GenerateRandomVLAN()
+
+	aEndAvailable, err := portSvc.CheckPortVLANAvailability(ctx, aEndUid, aEndVLAN)
+	if err != nil {
+		suite.FailNowf("cannot check a end vlan availability", "cannot check a end vlan availability %v", err)
+	}
+	if !aEndAvailable {
+		suite.FailNowf("a end vlan not available", "a end vlan %d is not available", aEndVLAN)
+	}
+
+	bEndAvailable, err := portSvc.CheckPortVLANAvailability(ctx, bEndUid, bEndVLAN)
+	if err != nil {
+		suite.FailNowf("cannot check b end vlan availability", "cannot check b end vlan availability %v", err)
+	}
+	if !bEndAvailable {
+		suite.FailNowf("b end vlan not available", "b end vlan %d is not available", bEndVLAN)
+	}
```

Comment:
This will create som flakey tests, I would select an available VLAN from the response or retry with a new one.